File: backend/dad-service-dashboard/src/main/java/com/dad/service/dashboard/entity/Chart.java
package com.dad.service.dashboard.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import com.dad.common.entity.BaseEntity;
import io.swagger.annotations.ApiModel;
import io.swagger.annotations.ApiModelProperty;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.time.LocalDateTime;

/**
 * Chart Entity
 */
@Data
@EqualsAndHashCode(callSuper = true)
@TableName("charts")
@ApiModel(value = "Chart", description = "Chart entity")
public class Chart extends BaseEntity {

    @TableId(type = IdType.AUTO)
    @ApiModelProperty(value = "Chart ID")
    private Long id;

    @ApiModelProperty(value = "Chart GUID")
    private String guid;

    @ApiModelProperty(value = "Dashboard ID")
    private Long dashboardId;

    @ApiModelProperty(value = "Chart name")
    private String name;

    @ApiModelProperty(value = "Description")
    private String description;

    @ApiModelProperty(value = "Chart type: LINE, BAR, PIE, etc.")
    private String chartType;

    @ApiModelProperty(value = "Data source ID")
    private Long dataSourceId;

    @ApiModelProperty(value = "Query configuration (JSON)")
    private String queryConfig;

    @ApiModelProperty(value = "Chart configuration (JSON)")
    private String chartConfig;

    @ApiModelProperty(value = "Style configuration (JSON)")
    private String styleConfig;

    @ApiModelProperty(value = "Interaction configuration (JSON)")
    private String interactionConfig;

    @ApiModelProperty(value = "Data cache seconds")
    private Integer dataCacheSeconds;

    @ApiModelProperty(value = "Position X")
    private Integer positionX;

    @ApiModelProperty(value = "Position Y")
    private Integer positionY;

    @ApiModelProperty(value = "Width (grid units)")
    private Integer width;

    @ApiModelProperty(value = "Height (grid units)")
    private Integer height;
}

File: frontend/src/api/chart.ts
import request from '@/utils/request'

export interface Chart {
  id?: number
  guid?: string
  dashboardId?: number
  name: string
  description?: string
  chartType: string
  dataSourceId: number
  queryConfig: string
  chartConfig: string
  positionX?: number
  positionY?: number
  width?: number
  height?: number
}

/**
 * Get chart by ID
 */
export function getChart(id: number) {
  return request({
    url: `/charts/${id}`,
    method: 'get'
  })
}

/**
 * Create chart
 */
export function createChart(data: Chart) {
  return request({
    url: '/charts',
    method: 'post',
    data
  })
}

/**
 * Update chart
 */
export function updateChart(id: number, data: Chart) {
  return request({
    url: `/charts/${id}`,
    method: 'put',
    data
  })
}

/**
 * Delete chart
 */
export function deleteChart(id: number) {
  return request({
    url: `/charts/${id}`,
    method: 'delete'
  })
}

/**
 * Get chart data
 */
export function getChartData(id: number) {
  return request({
    url: `/charts/${id}/data`,
    method: 'get'
  })
}

/**
 * Get charts by dashboard ID
 */
export function getChartListByDashboard(dashboardId: number) {
  return request({
    url: `/dashboards/${dashboardId}/charts`,
    method: 'get'
  })
}
